### PR TITLE
Add prompt utility

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+
+        while (true) {
+
+            final String input = reader.readLine();
+            if (StringUtils.isBlank(input)) {
+                if (hasDefaultValue || !isRequired) {
+                    return defaultValue;
+                }
+            }
+            if (Pattern.compile(regex).matcher(input).find()) {
```
Comment (Flanker32) [inline azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java]: What if `regex == null` ?

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+    }
+
+    public Boolean promoteYesNo(Boolean defaultValue, String message) throws IOException {
+        final String hintMessage = yesOrNo(defaultValue);
+        System.out.printf("%s %s: ", message, hintMessage);
+        System.out.flush();
+        while (true) {
+            final String input = reader.readLine();
+            if (StringUtils.isBlank(input)) {
+                return defaultValue;
```
Comment (Flanker32) [inline azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java]: defaultValue == null ?

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+            }
+            if (input.equalsIgnoreCase("Y")) {
+                return true;
+            }
+            if (input.equalsIgnoreCase("N")) {
+                return false;
+            }
+
+            System.out.printf("Invalid input(%s), %s %s: ", input, message, hintMessage);
```
Comment (Flanker32) [inline azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java]: We may split the prompt messages into two lines.
```
Invalid input(...), Confirm update the configuration? (Y/n)
```
```
Invalid input(...),
Confirm update the configuration? (Y/n)
```

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+            System.out.flush();
+        }
+    }
+
+    public <T> List<T> promoteManyEntities(String name, List<T> entities, Function<T, String> getNameFunc, boolean allowEmpty, String enterPromote,
```
Comment (Flanker32) [inline azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java]: Many - > Multi ?

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+            List<T> defaultList) throws IOException {
+        List<T> res = new ArrayList<>();
+        int index = 1;
+
+        if (!allowEmpty && entities.size() == 1) {
+            return entities;
+        }
+        System.out.println(String.format("Please select values for %ss:", name));
```
Comment (Flanker32) [inline azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java]: extra `s` or plural?

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+            System.out.println(displayLine);
+        }
+        final String hintMessage = String.format("(eg: %s, press %s %s)", TextUtils.blue("[1-2,4,4]"), TextUtils.blue("ENTER"), enterPromote);
+        System.out.print(String.format("Enter index values separated by comma%s: ", hintMessage));
+        System.out.flush();
+        for (;;) {
+
+            final String input = reader.readLine();
+            if (StringUtils.isBlank(input)) {
+                return defaultList;
```
Comment (Flanker32) [inline azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java]: null check?

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java [hunk 7]
```diff
@@ -0,0 +1,273 @@
...
+            }
+            if (isValidIntRangeInput(input)) {
+                for (final int i : parseIntRanges(input, MAX_ITEMS)) {
```
Comment (Flanker32) [inline azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java]: `Math.Min(MAX_ITEMS,entities.size()+1 )`?

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java [hunk 8]
```diff
@@ -0,0 +1,273 @@
...
+    public String promoteString(String name, String defaultValue, String regex, boolean isRequired) throws IOException {
+        final boolean hasDefaultValue = StringUtils.isNotBlank(defaultValue);
+        final String hintMessage = hasDefaultValue ? (" (" + TextUtils.green(defaultValue) + ")") : "";
+        System.out.printf("Please input the %s%s:", name, hintMessage);
+        System.out.flush();
+
+        while (true) {
+
+            final String input = reader.readLine();
+            if (StringUtils.isBlank(input)) {
```
Comment (Flanker32) [inline azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java]: I am not sure whether there will be a corner case is that user want to use an empty value but the default value is not empty?

Comment (andxu) [reply azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java]: The xml configuration in maven forbids empty string value.

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java (hunk 2 above)
Comment (andxu) [reply azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java]: press enter will keep original value or default value, if there is no original and default value, return null means get nothing.

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java (hunk 7 above)
Comment (andxu) [reply azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java]: good catch

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java (hunk 8 above)
Comment (andxu) [reply azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java]: add ":" to input empty string

**Pull request merged**